src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -370,6 +451,43 @@ static Log()
                    LogLevel.Debug,
                    EventIds.OnCircuitClosed,
                    "Closing circuit with id {CircuitId}.");
+
+                _beginInvokeDotNetStatic =
+                    LoggerMessage.Define<string, string, string>(
+                    LogLevel.Debug,
+                    EventIds.BeginInvokeDotNet,
+                    "Invoking static method '[{Assembly}]::{MethodIdentifier}' with callback id '{CallId}'");
+
+                _beginInvokeDotNetInstance =
+                LoggerMessage.Define<string, long, string>(
+                    LogLevel.Debug,
+                    EventIds.BeginInvokeDotNet,
+                    "Invoking instance method '{MethodIdentifier}' on instance '{DotNetObjectId}' with callback id '{CallId}'");
+
+                _endInvokeDispatchException = LoggerMessage.Define(
+                    LogLevel.Debug,
+                    EventIds.EndInvokeDispatchException,
+                    "There was an error invoking Microsoft.JSInterop.DotNetDispatcher.EndInvoke");
+
+                _endInvokeJSFailed = LoggerMessage.Define<long, string>(
+                    LogLevel.Debug,
+                    EventIds.EndInvokeJSFailed,
+                    "The JS interop call with '{AsyncCall}' failed with error '{Error}'.");
+
+                _endInvokeJSSucceeded = LoggerMessage.Define<long>(
+                    LogLevel.Debug,
+                    EventIds.EndInvokeJSSucceeded,
+                    "The JS interop call with '{AsyncCall}' succeeded.");
+
+                _dispatchEventFailedToParseEventDescriptor = LoggerMessage.Define(
+                    LogLevel.Debug,
+                    EventIds.DispatchEventFailedToParseEventDescriptor,
+                    "Failed to parse the event descriptor data when trying to dispatch an event.");
+
+                _dispatchEventFailedToDispatchEvent = LoggerMessage.Define(
+                    LogLevel.Debug,
+                    EventIds.DispatchEventFailedToDispatchEvent,
+                    "There was an error dispatching the event to the application.");
```

Comment:
Similar to in the comment at https://github.com/aspnet/AspNetCore/pull/12182/files#r303824668, could this be updated to include the event handler ID? This log message only gets used once we've successfully parsed the event descriptor, so we do have that ID. That will make various scenarios easier to diagnose.

Reply:
Updated